tests/agent_server/test_server_agent.py

```diff
@@ -1511,6 +1510,52 @@ async def assert_session_state(expected_agent_states: dict[str, AgentStatus], ex
     assert len(new_agent_processes) == 0, new_agent_processes
 
 
+@pytest.mark.parametrize("autostarted", (True, False))
+async def test_autostart_mapping_overrides_config(server, client, environment, async_finalizer, caplog, autostarted: bool):
```

Comment:
This test succeeds on master as well (see note in PR description) but I left it because it's a good check to have.